tests/main/interfaces-mount-control/task.yaml

```diff
@@ -36,30 +38,31 @@ execute: |
         snap connect "${SNAP_NAME}":mntctl
 
         echo "Verify that the snap can perform a mount"
-        mkdir -p "$MOUNT_DEST"
-        "${SNAP_NAME}".cmd mount -o bind,rw "$MOUNT_SRC" "$MOUNT_DEST"
+        mkdir -p "$MOUNT_DEST1"
+        "${SNAP_NAME}".cmd mount -o bind,rw "$MOUNT_SRC" "$MOUNT_DEST1"
 
         echo "Verify that the mount has been performed"
-        "${SNAP_NAME}".cmd grep "$MOUNT_DEST" /proc/self/mountinfo
+        "${SNAP_NAME}".cmd grep "$MOUNT_DEST1" /proc/self/mountinfo
 
         echo "and that it's only in the snap's namespace"
-        NOMATCH "$MOUNT_DEST" < /proc/self/mountinfo
+        NOMATCH "$MOUNT_DEST1" < /proc/self/mountinfo
 
         echo "Ensure that the mounted files are visible"
-        "${SNAP_NAME}".cmd test -e "$MOUNT_DEST/file1"
+        "${SNAP_NAME}".cmd test -e "$MOUNT_DEST1/file1"
 
         echo "Unmount via the system command umount(8)"
-        "${SNAP_NAME}".cmd umount "$MOUNT_DEST"
-        if "${SNAP_NAME}".cmd grep "$MOUNT_DEST" /proc/self/mountinfo; then
+        "${SNAP_NAME}".cmd umount "$MOUNT_DEST1"
+        if "${SNAP_NAME}".cmd grep "$MOUNT_DEST1" /proc/self/mountinfo; then
             echo "Unmount failed"
             exit 1
         fi
-        "${SNAP_NAME}".cmd test "!" -e "$MOUNT_DEST/file1"
+        "${SNAP_NAME}".cmd test "!" -e "$MOUNT_DEST1/file1"
 
         echo "Verify that a mount with a specific FS type can be created"
-        "${SNAP_NAME}".cmd mount -o rw -t tmpfs none "$MOUNT_DEST"
-        "${SNAP_NAME}".cmd grep "$MOUNT_DEST.*tmpfs" /proc/self/mountinfo
-        "${SNAP_NAME}".cmd umount "$MOUNT_DEST"
+        "${SNAP_NAME}".cmd mkdir -p "$MOUNT_DEST2"
+        "${SNAP_NAME}".cmd mount -o rw -t tmpfs none "$MOUNT_DEST2"
+        "${SNAP_NAME}".cmd grep "$MOUNT_DEST2.*tmpfs" /proc/self/mountinfo
+        "${SNAP_NAME}".cmd umount "$MOUNT_DEST2"
```

Comment:
Silly question, this PR is about adding `/` at the end but `$MOUNT_DEST2` is `$SNAP_COMMON/target2/path` without a trailing `/` - what am I missing :) ?

Reply:
It doesn't matter, I'm not sure whether the trailing `/` is added by the mount command, libc, or maybe even the kernel, but what matters is that the apparmor denial has it, even if it was not specified in the command line.

Reply:
yes, it seems we want a test that adds / as well explicitly

Reply:
It is not `util-linux`. The system calls either use trailing slash or not.

It is an apparmor. 

See [apparmor.d(5)](https://manpages.ubuntu.com/manpages/jammy/man5/apparmor.d.5.html), section "Globbing":

```
       When AppArmor looks up a directory the pathname being looked up will end with a slash
       (e.g., /var/tmp/); otherwise it will not end with a slash. Only rules that match a
       trailing slash will match directories. Some examples, none matching the /tmp/ directory
       itself, are:

       /tmp/*
           Files directly in /tmp.

       /tmp/*/
           Directories directly in /tmp.

       /tmp/**
           Files and directories anywhere underneath /tmp.

       /tmp/**/
           Directories anywhere underneath /tmp.
```

I suggest we do it correctly and always ensure we have a slash at the end of `mount` and `umount` rules. It could work   with `**` at the end. But so will `**/`. For all the other case, we always need a slash.

Reply:
And I meant always use the trailing slashing for the mount points. They are always directories.

Reply:
but we also support bind mounts on files in principle, not sure we have anything using that from mount-control. Not sure what apparmor expects in that case.

Reply:
Oh right!

Reply:
I've added a `/` in a couple of places; I didn't directly modify the variable `$MOUNT_DEST2` because it's also used for grepping `/proc/self/mountinfo`, where the trailing `/` is omitted.